### PR TITLE
Fixes #3962 - Refactor class overwriting

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by lzap** [comment]
Why do we override the class method?

**Comment by ares** [comment]
After this PR is merged we don't any more. The reason for this was a problem with STI and rails helpers which generates URL based on class name so e.g. for instance of Debian class it tried to call `debians_path` instead of `operatingsystems_path`

File: app/controllers/operatingsystems_controller.rb [hunk 1]
```diff
@@ -27,6 +27,7 @@ def edit
         @operatingsystem.os_default_templates.build(:template_kind_id => kind)
       end
     end if SETTINGS[:unattended]
+    @operatingsystem = generalize(@operatingsystem)
```
Comment (ohadlevy) [inline app/controllers/operatingsystems_controller.rb:30]: after filter maybe?

**Comment by ohadlevy** [comment]
minor comment, :+1: otherwise

File: app/controllers/operatingsystems_controller.rb (hunk 1 above)
Comment (ares) [inline app/controllers/operatingsystems_controller.rb:30]: After filters executes after rendering a view but we need this before rendering.

File: app/models/operatingsystems/archlinux.rb [hunk 2]
```diff
@@ -27,5 +22,9 @@ def url_for_boot(file)
   def display_family
     "Arch Linux"
   end
+
```
Comment (ohadlevy) [inline app/models/operatingsystems/archlinux.rb:25]: nitpick, extra two chars whitespace, will remove on merge

Comment (ares) [inline app/models/operatingsystems/archlinux.rb:25]: thank you :-)

**Comment by ohadlevy** [comment]
merged as 95a6671 thanks!

**Pull request closed without merging**